packages/hash/frontend/src/pages/[account-slug]/entities/new.page.tsx

```diff
@@ -0,0 +1,47 @@
+import { useRouter } from "next/router";
+import { useState } from "react";
+import { useEffectOnceWhen } from "rooks";
+import { useLoggedInUser } from "../../../components/hooks/useAuthenticatedUser";
+import { getPlainLayout, NextPageWithLayout } from "../../../shared/layout";
+import { mustBeVersionedUri } from "../types/entity-type/util";
+import { EntityPageLoadingState } from "./[entity-uuid].page/entity-page-loading-state";
+import { NewEntityPage } from "./[entity-uuid].page/new-entity-page";
+import { useCreateNewEntityAndRedirect } from "./[entity-uuid].page/shared/use-create-new-entity-and-redirect";
+
+const Page: NextPageWithLayout = () => {
+  const router = useRouter();
+  const { authenticatedUser } = useLoggedInUser();
+  const createNewEntityAndRedirect = useCreateNewEntityAndRedirect();
+  const [loading, setLoading] = useState(false);
+
+  useEffectOnceWhen(() => {
+    const init = async () => {
+      const entityTypeId = router.query["entity-type-id"] as string;
+
+      if (entityTypeId) {
+        try {
+          setLoading(true);
+          await createNewEntityAndRedirect(mustBeVersionedUri(entityTypeId));
+        } finally {
+          setLoading(false);
+        }
+      }
+    };
+
+    void init();
+  }, !!createNewEntityAndRedirect && !!authenticatedUser);
```

Comment:
This function is always defined, do you need it here?
```suggestion
  }, !!authenticatedUser);
```

Reply:
Addressed here https://github.com/hashintel/hash/pull/1492/commits/d83d521d7167144f024f073e894990567dfd4ed0